src/test/scala/wdl4s/types/WdlPairTypeSpec.scala

```diff
@@ -96,7 +122,49 @@ class WdlPairTypeSpec extends FlatSpec with Matchers {
         |  }
         |}
       """.stripMargin
-    
+
     noException should be thrownBy WdlNamespaceWithWorkflow.load(wdl, Seq.empty)
   }
+
+  it should "coerce a JsArray into a WdlArray of WdlPairs" in {
+    val jsArray = JsArray(
+      JsObject("Left" -> JsString("a"), "Right" -> JsNumber(1)),
+      JsObject("Left" -> JsString("b"), "Right" -> JsNumber(2)),
+      JsObject("Left" -> JsString("c"), "Right" -> JsNumber(3))
+    )
+
+    WdlArrayType(WdlPairType(WdlStringType, WdlIntegerType)).coerceRawValue(jsArray) match {
+      case Success(array) => array shouldEqual arrayOfPairs
+      case Failure(f) => fail(s"exception while coercing JsObject: $f")
+    }
+  }
+
+  it should "coerce a complex JsArray into a WdlArray of WdlPairs of WdlArrays" in {
+    val complexJsArray =  JsArray(
+      JsObject("Left" -> JsArray(JsString("a"), JsString("b")), "Right" -> JsArray(JsNumber(1), JsNumber(11))),
+      JsObject("Left" -> JsArray(JsString("c"), JsString("d")), "Right" -> JsArray(JsNumber(2), JsNumber(21))),
+      JsObject("Left" -> JsArray(JsString("e"), JsString("f")), "Right" -> JsArray(JsNumber(3), JsNumber(31)))
+    )
```

Comment:
ToL: Real JSON also has the added benefit of being almost pseudo-documentation in a TDD kind of a way.